pkg/controller/transfer/pvc.go

```diff
@@ -0,0 +1,171 @@
+package transfer
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	cdiv1 "kubevirt.io/containerized-data-importer/pkg/apis/core/v1beta1"
+	cdicontroller "kubevirt.io/containerized-data-importer/pkg/controller"
+)
+
+type pvcTransferHandler struct {
+	objectTransferHandler
+}
+
+func (h *pvcTransferHandler) ReconcilePending(ot *cdiv1.ObjectTransfer) (time.Duration, error) {
+	pvc := &corev1.PersistentVolumeClaim{}
+	pvcExists, err := h.reconciler.getSourceResource(ot, pvc)
+	if err != nil {
+		return 0, h.reconciler.setCompleteConditionError(ot, err)
+	}
+
+	if !pvcExists {
+		// will reconcile again when pvc is created/updated
+		if err := h.reconciler.setAndUpdateCompleteCondition(ot, corev1.ConditionFalse, "No source", ""); err != nil {
+			return 0, err
+		}
+
+		return 0, nil
+	}
+
+	if pvc.Spec.VolumeName == "" || pvc.Status.Phase != corev1.ClaimBound {
+		if err := h.reconciler.setAndUpdateCompleteCondition(ot, corev1.ConditionFalse, "PVC not bound", ""); err != nil {
+			return 0, err
+		}
+
+		return 0, nil
+	}
+
+	pods, err := cdicontroller.GetPodsUsingPVCs(h.reconciler.Client, pvc.Namespace, sets.NewString(pvc.Name), false)
+	if err != nil {
+		return 0, h.reconciler.setCompleteConditionError(ot, err)
+	}
+
+	if len(pods) > 0 {
```

Comment:
Just wondering if there is a race condition here between transferring and consuming a PVC?  What happens if we rebind the PVC while it's being used by a Pod in the source ns?

Reply:
That race cannot be eliminated as far as I know without changing kubelet.  As to what happens either the pod will be pending or the PVC will no get deleted.  The transfer controller will keep trying.